hikari/impl/entity_factory.py

```diff
@@ -1729,18 +1880,24 @@ def deserialize_message(  # noqa CFQ001 - Function too long
             mentions=NotImplemented,
         )
 
-        channels: typing.Mapping[snowflakes.Snowflake, channel_models.PartialChannel] = {}
         if raw_channels := payload.get("mention_channels"):
             channels = {c.id: c for c in map(self.deserialize_partial_channel, raw_channels)}
 
-        users: typing.Mapping[snowflakes.Snowflake, user_models.User] = {}
+        else:
+            channels = {}
+
         if raw_users := payload.get("mentions"):
             users = {u.id: u for u in map(self.deserialize_user, raw_users)}
 
-        role_ids: typing.Sequence[snowflakes.Snowflake] = []
```

Comment:
Why remove the typehints? They allow us to make sure that what is getting added to is what we expect

Reply:
was removing the building a list/dict twice also wouldn't passing them to the class's init cover what you described since these are only touched in one place